### ISSUE TITLE
Conditional execution of custom async tags

### ISSUE THREAD

**Issue by fabien** [mozilla/nunjucks#374]
Because the captured output will always be parsed and executed, it is currently impossible to implement a scenario where the async will only execute when required. Take for example a `{% cache %}` tag that would contain async tags like `{% fetch %}`. It's currently impossible to return cached content, without already running `fetch`, defeating the purpose.

**Comment by jlongster** [comment]
I don't understand what you mean exactly, are tags running inside `if` blocks when they shouldn't?

I see no reason why `fetch` could not look into the cache created by a `cache` tag.

**Comment by fabien** [comment]
Yes, IIRC async tags will always execute, even when the parent block shouldn't run. It'd be interesting to see how the scenario above would best be implemented:

```
{% cache %}
<h1>{{ datestamp() }}</h1>
{% fetch 123 %} // async
{% endfetch %}
{% endcache %}
```

**Comment by jlongster** [comment]
Ah, I can't remember has async tags are transformed, I'll take a look at it soon
